### PR TITLE
Massive TG middle-end port

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Ports alerts
Ports status effects
Ports traits
Ports movespeed modifiers

Massive code improvements

Credit to TG for all assets used.

## Changelog

:cl:
add: Ports TG alerts system. Alerts are the notifications that show up on the top-right when you're e.g. hot, hungry, a robot with a low power cell, etc.
add: Ports TG traits. Traits are a backend system that allows things to track what exactly is making you blind/clumsy/whatever.
add: Ports TG status effects. Status effects are things like being stunned, blinded, or having a speech impediment. We have these things, but they don't work very well at all (compared to how TG does em).
add: Ports TG movespeed modifiers. Movespeed modifiers allow code to cleanly adjust movement speed.
del: Removes old shitcode that was phased out with all the TG ports.
imageadd: Adds icons related to the new alerts system.
/:cl:

### CODE REVIEW

**Comment by TheDarkElites** [comment]
Dont forget to clean out any old code that provides the same or similar functionality

**Comment by aidan-bowman** [comment]
ready for speedmerge with 0 testing

**Comment by Tsurupeta** [comment]
fix your changelog my guy

**Comment by aidan-bowman** [comment]
> fix your changelog my guy

me after the lobotomy

File: code/modules/mob/living/carbon/human/memetic_insulation.dm [hunk 1]
```diff
@@ -106,7 +104,7 @@ Made by TheDarkElites
 	if(!(isobj(origin) || ismob(origin)))
 		return TRUE //if its not an object or mob it can always be identified/read (technically this should never happen but better safe than sorry)
 
-	var/viewdistance = 7 - get_how_nearsighted() //cant read if you're nearsighted and without prescription
+	var/viewdistance = is_nearsighted_currently() ? 7 : 2 //cant read if you're nearsighted
```
Comment (EgorDinamit) [inline code/modules/mob/living/carbon/human/memetic_insulation.dm]: This broke everything.
```suggestion
	var/viewdistance = is_nearsighted_currently() ? 2 : 7 //cant read if you're nearsighted
```

**Review by EgorDinamit — CHANGES_REQUESTED** [review]
This PR broke everything related to seeing &  hearing.
Fix it before TMing again.

File: code/modules/multiz/zmimic/mimic_movable.dm [hunk 2]
```diff
@@ -9,7 +9,7 @@
 		if (isturf(dest))
 			bound_overlay.forceMove(get_step(src, UP))
 			if (dir != bound_overlay.dir)
-				bound_overlay.set_dir(dir)
+				bound_overlay.setDir(dir)
```
Comment (Tsurupeta) [inline code/modules/multiz/zmimic/mimic_movable.dm:12]: yet again pizza proceeds to rename 1 proc for no reason and then change its name in 400 files

**Review by Tsurupeta — CHANGES_REQUESTED** [review]

**Comment by Tsurupeta** [comment]
also how about you split this shit, github doesnt even load files past ~half of this PR contents, I physically can't review unless I click "show file" on 250 files

**Comment by tichys** [comment]
someday this will be done

**Pull request merged**